Nodsoft.WowsReplaysUnpack/Controllers/ReplayControllerBase.cs

```diff
@@ -0,0 +1,242 @@
+﻿using Microsoft.Extensions.Logging;
+using Nodsoft.WowsReplaysUnpack.Core.Definitions;
+using Nodsoft.WowsReplaysUnpack.Core.Entities;
+using Nodsoft.WowsReplaysUnpack.Core.Extensions;
+using Nodsoft.WowsReplaysUnpack.Core.Models;
+using Nodsoft.WowsReplaysUnpack.Core.Network.Packets;
+using Nodsoft.WowsReplaysUnpack.Core.Security;
+using System.Reflection;
+
+namespace Nodsoft.WowsReplaysUnpack.Controllers;
+
+public abstract class ReplayControllerBase<T> : IReplayController
+	where T : class, IReplayController
+{
+	private static readonly Dictionary<string, MethodInfo[]> _methodSubscriptions;
+	private static readonly Dictionary<string, MethodInfo[]> _propertyChangedSubscriptions;
+	protected IDefinitionStore DefinitionStore { get; }
+	protected ILogger<Entity> EntityLogger { get; }
+
+	public UnpackedReplay Replay { get; protected set; }
+
+	static ReplayControllerBase()
+	{
```

Comment:
Consider to replace with source generation later if necessary due to performance